ios-base/Networking/APIService.swift

```diff
@@ -0,0 +1,180 @@
+//
+//  BaseService.swift
+//  ios-base
+//
+//  Created by Mauricio Cousillas on 6/4/19.
+//  Copyright © 2019 TopTier labs. All rights reserved.
+//
+
+import Foundation
+import Moya
+
+enum HTTPHeader: String {
+  case uid = "uid"
+  case client = "client"
+  case token = "access-token"
+  case expiry = "expiry"
+  case accept = "Accept"
+  case contentType = "Content-Type"
+}
+
+struct StatusCode {
+  static let unauthorized = 401
+}
+
+/**
+ Public protocol that defines the minimum API that an APIService should expose.
+ The APIService is the component in charge of handling al network request for
+ an specific TargetType.
+ Basic behaviour implemented using callbacks is provided by the BaseManager class.
+ */
+public protocol APIService {
+  /// The associated TargetType of the Service.
+  associatedtype ProviderType: TargetType
+  /// The MoyaProvider instance used to make the network requests.
+  var provider: MoyaProvider<ProviderType> { get }
+   /// The JSON decoding strategy used for JSON Responses.
+  var jsonDecoder: JSONDecoder { get }
+}
+
+/**
+ Base ApiService class that implements generic behaviour to
+ be extendended and used by subclassing it.
+ The base service has an associated TargetType, this means
+ that you should have **one and only one** manager for each TargetType.
+ This base implementation provides helpers to make requests with automatic
+ encoding if you provide a propper model as the expected result type.
+ */
+open class BaseApiService<T>: APIService where T: TargetType {
+  public typealias ProviderType = T
+
+  private var sharedProvider: MoyaProvider<T>!
+
+  /**
+   Default provider implementation as a singleton. It provides networking
+   loggin out of the box and you can override it if you want to add more middleware.
+   */
+  open var provider: MoyaProvider<T> {
+    guard let provider = sharedProvider else {
+      self.sharedProvider = MoyaProvider<T>(
+        plugins: [
+          NetworkLoggerPlugin(verbose: true, responseDataFormatter: JSONResponseDataFormatter)
+        ]
+      )
```

Comment:
self not needed, also the indentation is not the preferred. 

how about: 
`sharedProvider = MoyaProvider<T>(plugins: [networkLoggerPlugin])`
where networkLoggerPlugin is a computed property